rfcs/text/0013_saved_object_migrations.md

```diff
@@ -227,7 +204,7 @@ Note:
 1. Locate the source index by fetching aliases (including `.kibana` for versions prior to v7.10.0) `GET '/_alias/.kibana_current,.kibana_7.10.0,.kibana'`. The source index is the index the `.kibana_current` alias points to, or if it doesn’t exist, the index the `.kibana` alias points to.
 2. If `.kibana_current` and `.kibana_7.10.0` both exists and are pointing to the same index this version's migration has already been completed.
    1. Because the same version can have plugins enabled at any point in time, perform the mappings update in step (6).
-   2. If the target index contains any documents with an outdated migrationVersion number, log an error explaining that a re-migration is required and mark the saved object type these documents belong to as unavailable. See section (4.3.1.4).
+   2. If the target index contains any documents with an outdated migrationVersion number, log an error explaining that a re-migration is required and mark the saved object type these documents belong to as unavailable. See section (4.2.1.4).
```

Comment:
Would this also have the same consistency problems that tagging invalid objects has? Maybe it doesn't matter as much since these docs came from an old node.

EDIT: after reading the algo below, it's not immediately obvious how this situation could arise at all?

Reply:
This can happen when e.g.
1. 7.6.0 with spaces=enabled creates a default space document
2. User disables spaces plugin
3. User upgrades to 7.10.0 OSS (OSS has spaces=disabled so we don't migrate any spaces documents). 7.10.0 includes a migration for spaces documents
4. User notices their mistake and starts using 7.10.0 with x-pack and the spaces plugin enabled. The migration algorithm notices there's a spaces document with migrationVersion=7.6.0 instead of 7.10.0.

This does have a similar problem to tagging invalid objects in that if you accidentally had a plugin disabled like in the situation above you cannot rollback anymore you must re-migrate. If your re-migration fails you're stuck.

Reply:
I've updated 4.2.1.4 with this example and also changed the preferred way for dealing with orphan documents to prevent the data loss rollback scenario.